src/components/Tabs.js

```diff
@@ -47,19 +65,25 @@ class Tabs extends Component {
                 this.transformChildren(child.props && child.props.children, {
                     handleSelect,
                     selectedTab,
-                    firstLinkFound
+                    activeLinkStyle,
+                    name
                 })
             );
         });
     }
 
     render() {
         const handleSelect = this.props.handleSelect || this.handleSelect.bind(this);
-        const selectedTab = this.props.selectedTab || this.state.selectedTab;
+        const selectedTab = this.props.selectedTab ||
+            this.state.selectedTab ||
+            this.state.defaultTab ||
+            (this.state.defaultTab = this.findDefault(this.props.children));
```

Comment:
we should never mutate `this.state` directly (https://facebook.github.io/react/docs/component-api.html) 

> NEVER mutate this.state directly, as calling setState() afterwards may replace the mutation you made. Treat this.state as if it were immutable.

Can you refactor it to use `this.setState` call instead?